hw1_copy/noisifier.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+import random
+import librosa
+import os
+import argparse
+
+FORMATS = ['.wav', '.flac']
+SR = 25000
+
+
+def is_format(file):
+    flag = False
+    n = len(file)
+    for format in FORMATS:
+        flag  = flag or file[n - len(format):] == format
+    return flag
+
+
+def no_format(file):
+    n = len(file)
+    for format in FORMATS:
+        if file[n - len(format):] == format:
+            return file[:n - len(format)]
+    return "unknown"
+
+
+def load_audio_sec(file_path, sr=SR):
+    data, _ = librosa.core.load(file_path, sr)
+    if len(data) > sr:
+        data = data[:sr]
+    else:
+        data = np.pad(data, pad_width=(0, max(0, sr - len(data))), mode="constant")
+    return data
+
+
+def choose_noise(dir):
+    n = 0
+    for file in os.listdir(dir):
+        if is_format(file):
+            n += 1
+    i = random.randrange(n)
```

Comment:
this is very odd, please explore `sample` function

Reply:
Fixed